tests/indexes_/test_atlas_indexes.py

```diff
@@ -0,0 +1,204 @@
+        self.assertNotIn(
+            index.name,
+            connection.introspection.get_constraints(
+                cursor=None,
+                table_name=model._meta.db_table,
+            ),
+        )
+
+    def test_simple_vector_search(self):
+        with connection.schema_editor() as editor:
+            index = VectorSearchIndex(
+                name="recent_article_idx",
+                fields=["number"],
+            )
+            editor.add_index(index=index, model=Article)
+            self.assertAddRemoveIndex(editor, Article, index)
+
+    def test_multiple_fields(self):
+        with connection.schema_editor() as editor:
+            index = VectorSearchIndex(
+                name="recent_article_idx",
+                fields=["headline", "number", "body", "description_embedded"],
+            )
+            editor.add_index(index=index, model=Article)
+            index_info = connection.introspection.get_constraints(
+                cursor=None,
+                table_name=Article._meta.db_table,
+            )
+            expected_options = {
+                "latestDefinition": {
+                    "fields": [
+                        {"path": "headline", "type": "filter"},
+                        {"path": "number", "type": "filter"},
+                        {"path": "body", "type": "filter"},
+                        {
+                            "numDimensions": 10,
+                            "path": "description_embedded",
+                            "similarity": "cosine",
+                            "type": "vector",
+                        },
+                    ]
+                },
+                "latestVersion": 0,
+                "name": "recent_article_idx",
+                "queryable": False,
+                "type": "vectorSearch",
+            }
+            self.assertCountEqual(index_info[index.name]["columns"], index.fields)
+            index_info[index.name]["options"].pop("id")
+            index_info[index.name]["options"].pop("status")
+            self.assertEqual(index_info[index.name]["options"], expected_options)
+            self.assertAddRemoveIndex(editor, Article, index)
+
+    def test_field_not_exists(self):
+        index = VectorSearchIndex(
+            name="recent_article_idx",
+            fields=["headline", "non_existing_name", "title_embedded"],
+        )
+        msg = "Article has no field named 'non_existing_name'"
+        with self.assertRaisesMessage(FieldDoesNotExist, msg), connection.schema_editor() as editor:
+            editor.add_index(index=index, model=Article)
+
+    def test_field_size_required(self):
+        index = VectorSearchIndex(
+            name="recent_article_idx",
+            fields=["number_list"],
+        )
+        msg = "Atlas vector search requires size."
+        with self.assertRaisesMessage(ValueError, msg), connection.schema_editor() as editor:
+            editor.add_index(index=index, model=Article)
```

Comment:
For testing invalid model logic (which can hopefully be done as system checks), see
https://github.com/django/django/blob/922c1c732a47c02aa5ef28b0b1a2bd9bc9b92d87/tests/check_framework/test_model_checks.py